Tests/SwiftProtobufTests/Test_StreamDecodingIterator.swift

```diff
@@ -0,0 +1,58 @@
+// Tests/SwiftProtobufTests/Test_StreamDecodingIterator.swift - Delimited message tests
+//
+// Copyright (c) 2023 Apple Inc. and the project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See LICENSE.txt for license information:
+// https://github.com/apple/swift-protobuf/blob/main/LICENSE.txt
+//
+// -----------------------------------------------------------------------------
+
+import Foundation
+import XCTest
+import SwiftProtobuf
+
+class Test_StreamDecodingIterator: XCTestCase {
+  
+  func testEverything() {
```

Comment:
You should also have a test for the case of a zero-sized message (as Thomas pointed out above, zero-size messages are perfectly valid in protobuf).

Also tests for various truncation cases.  I would suggest have a suite of tests where you include short binary streams with known content (and problems).  See Test_AllTypes.swift for an example of this sort of thinking.

Among others:
* Empty stream with no bytes
* Stream with truncated varint
* Stream with zero varint and nothing else
* A stream with legal non-zero varint but no message
* Stream with valid varint, message, but the following varint is truncated

Reply:
Those tests have been added, plus one using a small buffer to test multiple-buffer message situation.